### PR TITLE
Small code structure improvements

### PR DESCRIPTION
**Pull request by rduenasf**
Hi @ebshimizu,

I did an initial take on refactoring the app a little bit. I worked on the following:

* Extract summary methods outside of `Class Database` as they are static methods.
* Extract database migration code into its own file
* Move format functions and math helpers into their own separate file
* Improve talent loading a little bit by using `require` instead of `readFile` and `JSON.parse`.
* Move some game data objects/arrays into separate files.
* Add Handlebars helpers `formatSeconds` and `formatPct` to improve code readability
* Reorganize `require`s in `js/index.js` 

My editor removes all leading spaces so that's why some files have a weird diff

### CODE REVIEW

**Comment by ebshimizu** [comment]
hey huge thanks for doing this!
I'll take some time over the weekend to look over the changes, but at a glance it looks like everything should merge cleanly.

File: js/heroes-talents.js [hunk 1]
```diff
@@ -162,10 +162,6 @@ class HeroesTalents {
     return this._roles[role].all.length;
   }
 
-  get heroCount() {
```
Comment (ebshimizu) [inline js/heroes-talents.js]: heroCount is used, don't remove. see https://github.com/ebshimizu/stats-of-the-storm/blob/master/js/hero-collection.js#L257

Comment (rduenasf) [reply js/heroes-talents.js]: was there twice :)

**Comment by rduenasf** [comment]
Awesome! :D 

Let's merge this first, I have more stuff but I think doing it in batches is better than doing it all at once. It's easier for you to review and for me in case you do some changes so I don't have to solve too many conflicts at once.

**Comment by ebshimizu** [comment]
sounds good! will keep an eye out for them.

**Pull request merged**